tests/Unit/Domain/FunctionsOfTime/Test_PiecewisePolynomial.cpp

```diff
@@ -179,6 +182,37 @@ void test_func_and_derivs() {
       quartic(std::array<DataVector, 5>{{c0, c1, c2, c3, c4}}));
 }
 
+void test_serialization_versioning() {
+  using Poly = FunctionsOfTime::PiecewisePolynomial<1>;
+  register_classes_with_charm<Poly>();
+  const std::unique_ptr<FunctionsOfTime::FunctionOfTime> poly(
+      new Poly(2.0, {{{3.0}, {4.0}}}, 5.0));
+  poly->update(5.0, {6.0}, 7.0);
+
+  // Because of the implementation-defined sign, there's no way to
+  // write char literals that won't cause narrowing errors.
+  const auto vector_char = [](auto... values) {
+    return std::vector<char>{static_cast<char>(values)...};
+  };
+
+  const auto serialization_v2 = vector_char(
```

Comment:
Maybe add a note how this data representation can be obtained when the test is updated? I think this kind of test is a great idea, and it would be nice to make it easy for people to add these tests to more domain-related classes.